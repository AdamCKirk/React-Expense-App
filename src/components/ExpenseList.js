import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectedExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p> No Expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />
                })
            )
        }
        
    </div>
);

// Map the infomation to the state
const mapStateToProps = (state) => {
    return {
        expenses: selectedExpenses(state.expenses, state.filters)
    };
}

export default connect(mapStateToProps)(ExpenseList);
