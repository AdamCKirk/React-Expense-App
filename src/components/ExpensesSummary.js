import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectedExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpensesSumamry = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
    const formattedExpeneseTotal = ' £' + numeral( expensesTotal / 100).format('0,0.00')

    return (
        <div>
            <h3>
                Viewing {expenseCount} {expenseWord} totalling {formattedExpeneseTotal}
            </h3>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
}

export default connect(mapStateToProps)(ExpensesSumamry);

// const ExpensesSumamry = (props) => (
//     <div>
//         <p>
//             Viewing {props.expenses.length + ' '} 
//             expenses totalling 
//             {' £' + numeral( selectExpensesTotal(props.expenses) / 100).format('0,0.00')}
//         </p>
//     </div>
// )

// const mapStateToProps = (state) => {
//     return {
//         expenses: selectedExpenses(state.expenses, state.filters)
//     };
// }

// export default connect(mapStateToProps)(ExpensesSumamry);