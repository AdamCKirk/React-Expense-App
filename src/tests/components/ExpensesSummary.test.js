import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSumamry } from '../../components/ExpensesSummary';

test('should render correctly ExpensesSummary with 1 expenses', () => {
    const wrapper = shallow(<ExpensesSumamry expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot();

});

test('should render correctly ExpensesSummary with many expenses', () => {
    const wrapper = shallow(<ExpensesSumamry expenseCount={23} expensesTotal={23512340987} />)
    expect(wrapper).toMatchSnapshot();
});
