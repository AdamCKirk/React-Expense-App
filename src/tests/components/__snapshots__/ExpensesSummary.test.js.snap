// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ExpensesSummary with 1 expenses 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
     £2.35
  </h3>
</div>
`;

exports[`should render correctly ExpensesSummary with many expenses 1`] = `
<div>
  <h3>
    Viewing 
    23
     
    expenses
     totalling 
     £235,123,409.87
  </h3>
</div>
`;
